qemu/glib_compat.c

```diff
@@ -0,0 +1,761 @@
+/*
+glib_compat.c replacement functionality for glib code used in qemu
+Copyright (C) 2016 Chris Eagle cseagle at gmail dot com
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "glib_compat.h"
+
+#ifdef _WIN32
+   #ifdef _WIN64
+      #define __HAVE_64_BIT_PTRS
```

Comment:
do we need these macros __HAVE_64_BIT_PTRS & __HAVE_32_BIT_PTRS? i cant find where they are used